core/database_api.php

```diff
@@ -1323,3 +1323,23 @@ function db_mysql_fix_utf8( $p_string ) {
 function db_empty_result() {
 	return new ADORecordSet_empty();
 }
+
+/**
+ * Returns the maximum value of a valid integer for current driver, for the integer
+ * type that is actually used by Adodb for translating "I UNSIGNED", which is
+ * the type used in mantis schema for most numeric id fields.
+ * Adodb maps "I" as "I4". The actual types used for each driver are mapped in
+ * ADODB_DataDict objects, function ActualType.
+ * @return int
+ */
+function db_max_uint() {
+	# Safe assumption is for maximum value of 4 bytes signed integer: 2^31-1
+	# This is the limit for databases that dont support unsigned types, eg pgsl, mssql
+	$t_max = 2147483647;
+
+	if( db_is_mysql() ) {
```

Comment:
> id's > 2147483647 is not a real use case 

good enough, if we take that compromise, that can be a universal limit to check against.
If we rely in Adodb mapping to INT4 types, worst case is that the DB does not provide unsigned types, so 31bits is a common upper limit.

side-note: 2038 date limit?


> I would just use a constant for it.

probably better for performance. 
Show me one more up for this and i will change to a constant, @vboctor, @dregad  ?